src/context.ts

```diff
@@ -17,16 +17,21 @@ export function createContext<Extensions = any, EmitOptions = any>(props: Create
 
   const hooks = props.adapter?.(emit).hooks
 
-  function emit<P>(event: Eventa<P>, payload: P, options?: EmitOptions) {
+  function emit<P>(event: Eventa<P>, payload: P, options?: EmitOptions): boolean {
     const emittingPayload = { ...event, body: payload }
+    let isConsumed = false
```

Comment:
I am afraid of this may not work as expected since if we are running electron & worker in different threads, the reference to how `listener` was called are different to how the `emit` in `defineInvoke` could get. 

I think we might need to add one more event to `InvokeEventa` and call it as `connected` or `consuming`, then at least the peer that `emit`(s) the event would be able to be aware about whether there is a remote peer consuming the request. Along with the `connected` or `consuming`, we can add one more option called `comsumingTimout` works like how TCP timeout was. (maybe we could think about how we would work with UDP too.)

PS: oh and we could have a UDP adapter!